### PR TITLE
V6

### PR DESCRIPTION
**Pull request by danielbarion**
## v6

The v6 is a version compatible with Pixi.js v8, also we are re-organizing some information about the project (like examples, typescript types, ...).

### TODO's

- [x] Pixi v8 added as peerDependencies
- [x] Readme updated with versions compatible with Pixi.js versions
- [x] typescript types on ease.ts
- [x] `yarn` changed to `npm run` on scripts - we still can use yarn on our local to develop it as normal :) 

### Pull Requests merged in V6

- https://github.com/pixi-viewport/pixi-viewport/pull/463
- https://github.com/pixi-viewport/pixi-viewport/pull/420

### CODE REVIEW

**Comment by danielbarion** [comment]
I'm checking the lint issue

**Comment by danielbarion** [comment]
PR tested locally on my game project using `yalc` 👍🏻

File: src/plugins/Clamp.ts [hunk 1]
```diff
@@ -22,45 +22,45 @@ import type { Viewport } from '../Viewport';
...
+   * @default false
+   */
     top?: number | boolean | null;
 
     /**
-     * Clamp right; true = viewport.worldWidth
-     *
-     * @default false
-     */
+   * Clamp right; true = viewport.worldWidth
```
Comment (danielbarion) [inline src/plugins/Clamp.ts:39]: @davidfig my prettier is not recognizing the eslint, so when I save the files, it gets formated totally differently (all files I save) - so before submitting the PR I run `yarn eslint:fix` to apply the correct linting, and that was the result

File: src/global.d.ts [hunk 2]
```diff
@@ -3,19 +3,19 @@
 declare namespace GlobalMixins
```
Comment (Zyie) [inline src/global.d.ts:3]: hey @danielbarion I believe this should be `PixiMixins` for v8

Comment (danielbarion) [reply src/global.d.ts:3]: @Zyie like that?

image

File: src/plugins/Clamp.ts (hunk 1 above)
Comment (davidfig) [reply src/plugins/Clamp.ts:39]: ok, no problem

**Review by davidfig — APPROVED** [review]

File: src/global.d.ts (hunk 2 above)
Comment (danielbarion) [reply src/global.d.ts:3]: @Zyie looks like @davidfig didn't see this thread, but please let me know what the correct types is and I'll create a PR and merge for 6.0.1

Comment (Zyie) [reply src/global.d.ts:3]: hey @danielbarion yeah this looks correct

**Pull request merged**